react-cors-spa/src/ProductDialogue.jsx

```diff
@@ -0,0 +1,93 @@
+import { useQuery } from '@tanstack/react-query';
+import { useForm } from 'react-hook-form';
+import fetchCategories from './services/CategoriesService';
+import "./ProductDialogue.css";
+
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+import TextField from '@mui/material/TextField';
+import MenuItem from '@mui/material/MenuItem';
+import Select from '@mui/material/Select';
+import FormControl from '@mui/material/FormControl';
+import { Button, Typography, Box } from "@mui/material";
+import CloseOutlinedIcon from '@mui/icons-material/CloseOutlined';
+
+function ProductDialogue({ onDialogue }) {
+    // Populate categories from DB with useQuery
+    const { data } = useQuery({
+        queryKey: ['categoryData'],
+        queryFn: fetchCategories,
+        select: (data) => {
+            return data?.data?.sort((a, b) => {
+                const nameA = a.name.toUpperCase();
+                const nameB = b.name.toUpperCase();
+                if (nameA < nameB) {
+                    return -1;
+                }
+                if (nameA > nameB) {
+                    return 1;
+                }
+                
+                return 0;
+                });        
+        }
+    });
+
+    // Verify form completion with formState
+    const { register, handleSubmit, formState: {errors} } = useForm({
+        defaultValues: {
+            Name: "",
+            SKU: "",
+            Size: "",
+            Description: "",
+            Category: "Category",
+        }
+    });
+
+    return(
+        <Dialog open>
+        <Box className='titleAndButton'>
+            <DialogTitle>Add Product</DialogTitle>
+            <CloseOutlinedIcon className='exitButton' onClick={onDialogue}></CloseOutlinedIcon>
```

Comment:
onDialogue is kinda of a vague name. onDialogClose?

Reply:
Makes sense and works for me